src/components/Comments.js

```diff
@@ -40,27 +43,67 @@ const Comments = (props) => {
 
   const [viewRepliesToggled, setViewRepliesToggled] = useState([])
 
+  const [attachments, setAttachments] = useState([])
+
+  const [replyAttachments, setReplyAttachments] = useState([])
+
   useEffect(() => {
-    const fetchComments = async () => {
+    const fetchStoryComments = async () => {
       const response = await userStory.getComments(storyId)
       setComments(response.data.data.userStory.user_story_comments)
     }
-    fetchComments()
+    fetchStoryComments()
   }, [storyId, fetchComments])
 
+  useEffect(
+    () => () => {
+      attachments.forEach((file) => URL.revokeObjectURL(file.preview))
+    },
+    [attachments]
+  )
+
+  useEffect(
+    () => () => {
+      replyAttachments.forEach((file) => URL.revokeObjectURL(file.preview))
+    },
+    [replyAttachments]
+  )
+
   const addComment = async (data) => {
-    const response = await userStory.postComment(data.addComment, storyId, id)
-    setComments([
-      ...comments,
-      response.data.data.createUserStoryComment.userStoryComment
-    ])
+    const formData = new FormData()
+    data.user = id
+    data.user_story = storyId
+    formData.append('data', JSON.stringify(data))
+
+    if (attachments.length) {
```

Comment:
Lets move this code to a function `addAttachments` which can take `attachments` and `formData` as parameters. The same function can be used for adding attachments for comment replies.